File: oop/heritage/Java/exercice4/src/Forgeron.java


public class Forgeron extends Personne {
    private String metier;

    public Forgeron() {
        super();
    }

    public Forgeron(String prenom) {
        super(prenom);
        metier = "Forgeron";
    }

    @Override
    public void affiche() {
        super.affiche();
        System.out.print(" le " + this.metier);
    }
}

File: oop/heritage/CPP/exercice4/src/Forgeron.cpp
#include "Forgeron.hpp"

Forgeron::Forgeron() : Personne(), metier("") {}

Forgeron::Forgeron(const std::string &prenom) : Personne(prenom), metier("Forgeron") {}

void Forgeron::affiche() const
{
    Personne::affiche();
    std::cout << " le " << metier;
}
